R/shinyalert.R

```diff
@@ -253,10 +265,34 @@ shinyalert <- function(
       }
     }, once = TRUE)
     params[['callbackR']] <- NULL
+  } else {
+    # Still create `cbid` used by closeAlert to explicitly close this alert
```

Comment:
In the if statement I added a comment that the serialization isn't happening to prevent a performance issue https://github.com/daattali/shinyalert/issues/19

Did you notice that and test if this change is going to re-introduce that issue?

Reply:
I don't recommend `params <- as.list(environment())` at the very beginning. Chances are too small for two alerts sharing the same cbid with `runif` postfix. If you get rid of the enclosing environment, `digest::digest` will be quite fast as everything else should be basic types and contain no environment. Alternatively you could use `utils::capture.output({print(environment())})` to distinguish environments because the runtime environment of R functions will unlikely to be named, meaning its memory address will be printed, resulting in something like `"<environment: 0x7f802d60a1b0>"`. Serializing string is much faster than serializing an environment.

Reply:
Or, if you do want to keep `params <- as.list(environment())`, you can create random `cbid` with no digests for alerts without callbacks. The returned `cbid` can be used to close that specific alert or writers can use that `cbid` to generate their own callbacks outside of `shinyalert`.

Reply:
The line `params <- as.list(environment())` is not for serialization or creating a unique ID, it was simply a way to get a list of all the function arguments values - default values for arguments that weren't provided and the user-supplied values for the arguments that were provided. Do you know of an alternative way to get all the arguments of the current function as a list?

But anyway, you're right that the digest isn't actually required , it was only to add some randomness. Instead of using the params hash plus a random integer, it might be better to just use the `uuid` package

Reply:
In that way you are right, but your function arguments contains no function except `callbackR`, which will be removed before serialization. Therefore serialization should be fast. Also personally I think using `runif` is good enough to create unique IDs unless people set seed before calling the function. In that case, you might want to consider `utils::capture.output({print(environment())})` I proposed to ensure different IDs created.